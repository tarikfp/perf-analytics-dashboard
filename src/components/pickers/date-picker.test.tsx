import { createWithThemeContext, renderWithThemeContext } from '@utils';
import * as React from 'react';
import DatePickers from './date-picker';

// Date pickers have been tested internally by the mui pickers community.
// @see https://github.com/mui/material-ui-pickers/blob/next/lib/src/__tests__/DatePickerTestingLib.test.tsx

const props = {
  startDate: new Date(),
  endDate: new Date(),
  onStartDateChange: jest.fn(),
  onEndDateChange: jest.fn(),
  onSubmit: jest.fn(),
};

test('date pickers renders correctly', () => {
  const tree = createWithThemeContext(<DatePickers {...props} />);
  expect(tree).toMatchSnapshot();
});

test('submit button renders correctly', () => {
  const { getByText } = renderWithThemeContext(<DatePickers {...props} />);
  expect(getByText('Submit')).toBeTruthy();
});
