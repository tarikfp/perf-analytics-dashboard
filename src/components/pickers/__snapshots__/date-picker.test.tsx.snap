// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date pickers renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 4,
      "height": 120,
      "marginTop": 20,
      "padding": 10,
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-125lwsu-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1evdibm-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r0:"
          id=":r0:-label"
        >
          Start date
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            aria-label="Choose date, selected date is May 3, 2022"
            aria-readonly={true}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="05/03/2022 02:40 pm"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                Start date
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1ube6tp-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r2:"
          id=":r2:-label"
        >
          End date
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            aria-label="Choose date, selected date is May 3, 2022"
            aria-readonly={true}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r2:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="05/03/2022 02:40 pm"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                End date
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-md-4 css-jgkidt-MuiGrid-root"
    >
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1qicx88-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Submit
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
